modules/support_ticket/src/Access/SupportTicketPreviewAccessCheck.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\support_ticket\Access\SupportTicketPreviewAccessCheck.
+ */
+
+namespace Drupal\support_ticket\Access;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\support_ticket\SupportTicketInterface;
+
+/**
+ * Determines access to support_ticket previews.
+ *
+ * @ingroup support_ticket_access
+ */
+class SupportTicketPreviewAccessCheck implements AccessInterface {
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructs a EntityCreateAccessCheck object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager) {
+    $this->entityManager = $entity_manager;
+  }
+
+  /**
+   * Checks access to the support_ticket preview page.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The currently logged in account.
+   * @param \Drupal\support_ticket\SupportTicketInterface $support_ticket_preview
+   *   The support_ticket that is being previewed.
+   *
+   * @return string
+   *   A \Drupal\Core\Access\AccessInterface constant value.
+   */
+  public function access(AccountInterface $account, SupportTicketInterface $support_ticket_preview) {
```

Comment:
@generic Except for the Interface and the hardcoded 'support_ticket', this is generic functionality.